project_electron/rights/views.py

```diff
@@ -4,22 +4,77 @@
+class RightsManageView(RACAdminMixin, CreateView):
     template_name = 'rights/manage.html'
     model = RightsStatement
     form_class = RightsForm
 
-class RightsDetailView(RACAdminMixin, DetailView):
+    def get_context_data(self, *args, **kwargs):
+        context = super(RightsManageView, self).get_context_data(**kwargs)
+        context['basis_form'] = RightsForm()
+        context['copyright_form'] = CopyrightFormSet()
+        context['license_form'] = LicenseFormSet()
+        context['statute_form'] = StatuteFormSet()
+        context['other_form'] = OtherFormSet()
+        context['organization'] = Organization.objects.get(pk=self.request.GET.get('org'))
+        return context
+
+    def post(self, request, *args, **kwargs):
+        form = RightsForm(request.POST)
+        rights_form = form.save(commit=False)
+        rights_form.organization = Organization.objects.get(pk=request.GET.get('org'))
+        if rights_form.rights_basis == 'Copyright':
+            formset = CopyrightFormSet(request.POST, instance=rights_form)
+            formset_key = 'copyright_form'
+        elif rights_form.rights_basis == 'License':
+            formset = LicenseFormSet(request.POST, instance=rights_form)
+            formset_key = 'license_form'
+        elif rights_form.rights_basis == 'Statute':
+            formset = StatuteFormSet(request.POST, instance=rights_form)
+            formset_key = 'statute_form'
+        else:
+            formset = OtherFormSet(request.POST, instance=rights_form)
+            formset_key = 'other_form'
+        if formset.is_valid():
+            rights_form.save()
+            formset.save()
+            return redirect('rights-grants', rights_form.pk)
+        else:
+            return render(request,'rights/manage.html', {formset_key: formset, 'basis_form': form})
+
+class RightsGrantsManageView(RACAdminMixin, CreateView):
+    template_name = 'rights/manage.html'
+    model = RightsStatement
+    form_class = RightsForm
+
+    def get_context_data(self, *args, **kwargs):
+        context = super(RightsGrantsManageView, self).get_context_data(**kwargs)
+        context['granted_formset'] = RightsGrantedFormSet()
+        context['rights_statement'] = RightsStatement.objects.get(pk=self.kwargs.get('pk'))
+        context['rights_basis_info'] = context['rights_statement'].get_rights_info_object
+        context['organization'] = context['rights_statement'].organization
+        return context
+
+    def post(self, request, *args, **kwargs):
+        rights_statement = RightsStatement.objects.get(pk=self.kwargs.get('pk'))
+        formset = RightsGrantedFormSet(request.POST, instance=rights_statement)
+        if formset.is_valid():
+            formset.save()
+            return redirect('rights-detail', self.kwargs.get('pk'))
+        else:
+            return render(request,'rights/manage.html', {'granted_formset': formset})
+
+class RightsDetailView(DetailView):
```

Comment:
This makes this public.. We should really talk about a parent class that at least passes loggedinmixin.. But also this may be the second place i found that doesn't have logic to isolates view to org only.